packages/drivers/odsp-driver/src/odspDocumentServiceFactoryCore.ts

```diff
@@ -80,6 +80,25 @@ export class OdspDocumentServiceFactoryCore implements IDocumentServiceFactory {
             throw new Error("File path should be provided!!");
         }
 
+        let fileInfo: INewFileInfo | IExistingFileInfo;
+        let createShareLinkParam: ShareLinkTypes | ISharingLinkKind | undefined;
+        if (odspResolvedUrl.fileName) {
+            createShareLinkParam = getSharingLinkParams(this.hostPolicy, searchParams);
+            fileInfo = {
+                driveId: odspResolvedUrl.driveId,
+                siteUrl: odspResolvedUrl.siteUrl,
+                filePath,
+                filename: odspResolvedUrl.fileName,
+                createLinkType: createShareLinkParam,
+            };
+        } else if (odspResolvedUrl.itemId) {
```

Comment:
where does this happen? how does the itemId make it on the resolved url

Reply:
When `createNew` DriverHeader is not provided, OdspDriverUrlResolver will take the itemId from the IRequest URL. 

i.e., At application side, I'm creating URL like this:

```
    const odspCreateContainerRequest: IRequest = {
      url: createOdspUrl({
        siteUrl: location.siteUrl,
        driveId: location.driveId,
        itemId: location.itemId!,
        dataStorePath: '/'
      })
    };

    return loader.createDetachedContainer({ package: 'no-dynamic-package', config: {} }).then((container) => {
      return container.attach(odspCreateContainerRequest).then(() => container);
    });
```